### PR TITLE
Closet Destruction Contents Ejection

### PR DESCRIPTION
**Pull request by SiegeB0t**
## About The Pull Request
Explosions, acid, and plasma cutters now release the contents of a locker or crate.

## Why It's Good For The Game
Punishes bad grenade placement.
Makes locker/explosive interactions more interesting.
Acid does not destroy all contents wholesale.
Prevents PC users from qdeleting their cowardly comrades hiding in lockers.

## Changelog
:cl:
balance: Makes unlocked lockers/crates more vulnerable to explosions.
balance: Xeno acid dump does not destroy the contents of whatever is melted.
fix: Plasma cutter now deconstructs objects rather than simply deleting them. 
/:cl:

### CODE REVIEW

**Comment by Losenis** [comment]
Make it so that it also applies to acid, xenos can delete the entirety of a locker's contents by acid in the same way.

**Comment by Naaanii** [comment]
Crates having this would also be nice.

**Comment by SiegeB0t** [comment]
This applies to crates as well. "About The Pull Request" updated.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 1]
```diff
@@ -156,30 +156,27 @@
...
-// this should probably use dump_contents()
 /obj/structure/closet/ex_act(severity)
+	var/dmg
 	switch(severity)
 		if(EXPLODE_DEVASTATE)
-			for(var/am in contents)//pulls everything out of the locker and hits it with an explosion
-				var/atom/movable/movable_content = am
-				movable_content.forceMove(loc)
-				movable_content.ex_act(severity)
+			contents_explosion(EXPLODE_HEAVY)
```
[the remark this answers is not in the document]
Comment (SiegeB0t) [reply code/game/objects/structures/crates_lockers/closets.dm:163]: I thought it would be funny but I have no justification for it. so...

**Review by TiviPlus — APPROVED** [review]

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**